lib/rules/no-empty.js

```diff
@@ -14,20 +15,39 @@ module.exports = function(context) {
     return {
 
         "BlockStatement": function(node) {
-            var ancestors = context.getAncestors(),
-                parent = ancestors[ancestors.length - 1],
-                parentType = parent.type,
-                hasHandler = !!(parent.handler || (parent.handlers && parent.handlers.length)),
-                isFinallyBlock = (parentType === "TryStatement") && (parent.finalizer === node);
-
-            if (/Function|CatchClause/.test(parentType) ||
-                    (isFinallyBlock && !hasHandler)) {
+            var ancestors,
+                parent,
+                parentType,
+                isFinallyBlock,
+                comments,
+                i;
+
+            // if the body is not empty, we can just return immediately
+            if (node.body.length !== 0) {
                 return;
             }
 
-            if (node.body.length === 0) {
-                context.report(node, "Empty block statement.");
+            ancestors = context.getAncestors();
```

Comment:
I actually kept this from your original code but clearly only using node.parent seems more efficient.